cfme/tests/infrastructure/test_vm_migrate.py

```diff
@@ -74,7 +76,33 @@ def test_vm_migrate_should_create_notifications_when_migrations_fail():
         initialEstimate: 1/4h
         startsin: 5.10
         tags: service
+        testSteps:
+            1. add provider to appliance
+            2. create vm
+            3. migrate vm, use same host and datastore is None
+        expectedResults:
+            1.
+            2.
+            3. check error on UI and in the log file.
     Bugzilla:
         1478462
     """
-    pass
+    view = navigate_to(create_vm, 'Details')
+    vm_host = view.entities.summary('Relationships').get_text_of('Host')
+    hosts = [vds.name for vds in provider.hosts.all() if vds.name in vm_host]
+    migrate_to = hosts[0]
+    err_msg = (f'The VM \'{create_vm.name}\' can not be migrated to the same host it is '
+               'already running on.]')
+    evm_tail = LogValidator('/var/www/miq/vmdb/log/automation.log',
+                            matched_patterns=[err_msg],
+                            hostname=appliance.hostname)
+    evm_tail.start_monitoring()
+    create_vm.migrate_vm("email@xyz.com", "first", "last", host=migrate_to, datastore="<None>")
+    request_description = create_vm.name
+    cells = {'Description': request_description, 'Request Type': 'Migrate'}
+    migrate_request = appliance.collections.requests.instantiate(request_description, cells=cells,
+                                                                 partial_check=True)
+    migrate_request.wait_for_request(method='ui')
+    assert not migrate_request.is_succeeded(method='ui'), "VM migration does not failed"
+    assert err_msg in migrate_request.row.last_message.text, migrate_request.row.last_message.text
+    assert evm_tail.validate(wait="30s"), "VM migrate failed ERROR not found in log file"
```

Comment:
Can you use the context manager for `LogValidator` here instead? https://github.com/ManageIQ/integration_tests/blob/734fd6ec093cf9040709744cd69001752cb3cf53/cfme/utils/log_validator.py#L59